lib/cell.rb

```diff
@@ -21,24 +21,43 @@ def fired_upon?
     @shot_fired
   end
 
+  # def fire_upon
+  #   if @ship
+  #     @ship.hit
+  #     @shot_fired = true
+  #   else
+  #     @shot_fired = true
+  #   end
+  # end
+
   def fire_upon
-    if @ship != nil
-      @ship.hit
-      @shot_fired = true
-    else
-      @shot_fired = true
-    end
+    @ship&.hit
```

Comment:
After finishing iteration 1 we looked at methods we could refactor/ make more DRY. Allen discovered this safe navigation operator that replaces line 25-26.